prow/config/config.go

```diff
@@ -2451,12 +2459,27 @@ func setBrancherRegexes(br Brancher) (Brancher, error) {
 }
 
 func setChangeRegexes(cm RegexpChangeMatcher) (RegexpChangeMatcher, error) {
-	if cm.RunIfChanged != "" {
-		re, err := regexp.Compile(cm.RunIfChanged)
-		if err != nil {
-			return cm, fmt.Errorf("could not compile run_if_changed regex: %v", err)
+	reprop := []struct {
```

Comment:
Nit: While I understand the desire of de-duplicating code, this is 21 LOC  and an additional indirection compared to 14 LOC it would take to have the check twice, once per field :)

Reply:
I agree. LMK when done reviewing and I'll fix it up.

Reply:
I couldn't stand *not* de-duplicating, so I came up with this:

```go
	var reString, propName string
	if reString = cm.RunIfChanged; reString != "" {
		propName = "run_if_changed"
	} else if reString = cm.SkipIfOnlyChanged; reString != "" {
		propName = "skip_if_only_changed"
	}
	if reString != "" {
		re, err := regexp.Compile(reString)
		if err != nil {
			return cm, fmt.Errorf("could not compile %s regex: %v", propName, err)
		}
		cm.reChanges = re
	}
	return cm, nil
```

WDYT?

Reply:
Did this.